cmd/virt-launcher/virt-launcher.go

```diff
@@ -432,6 +447,10 @@ func main() {
 			*gracePeriodSeconds,
 			shutdownCallback)
 
+		// waits until virt-vnc socket is ready and than mark its parent folder as read only
+		// workaround preventing QEMU from deleting socket prematurely
+		writeProtectPrivateDir(*uid)
```

Comment:
this function waits forever for the socket to appear. I'd feel better if there was a timeout, or we just executed this as a goroutine in the background so it doesn't potentially block the monitor function.

Reply:
You are right. I have added 20s as max limit and started is as a goroutine.
It will not block.